### ISSUE TITLE
rmtree called on a symlink fails

### ISSUE THREAD

**Issue by skyreflectedinmirrors** [AMDResearch/omniperf#296]
Bug report from a hackathon in:  https://github.com/AMDResearch/omniperf/blob/79260da3354626c43452dfe27ca5dea0e8d833a1/src/omniperf_soc/soc_base.py#L208

with:

```
ROC Profiler: /opt/rocm-5.7.1/bin/rocprof
Execution mode = profile

  ___                  _                  __ 
 / _ \ _ __ ___  _ __ (_)_ __   ___ _ __ / _|
| | | | '_ ` _ \| '_ \| | '_ \ / _ \ '__| |_ 
| |_| | | | | | | | | | | |_) |  __/ |  |  _|
 \___/|_| |_| |_|_| |_|_| .__/ \___|_|  |_|  
                        |_|                  

Profiler choice = rocprofv1
Traceback (most recent call last):
  File "/autofs/nccs-svm1_sw/crusher/amdsw/omniperf/2.0.0-TP1/bin/omniperf", line 50, in <module>
    main()
  File "/autofs/nccs-svm1_sw/crusher/amdsw/omniperf/2.0.0-TP1/bin/omniperf", line 39, in main
    omniperf.run_profiler()
  File "/autofs/nccs-svm1_sw/crusher/amdsw/omniperf/2.0.0-TP1/bin/utils/utils.py", line 46, in wrap_function
    result = function(*args, **kwargs)
  File "/autofs/nccs-svm1_sw/crusher/amdsw/omniperf/2.0.0-TP1/bin/omniperf_base.py", line 248, in run_profiler
    self.__soc[self.__mspec.gpu_arch].profiling_setup()
  File "/autofs/nccs-svm1_sw/crusher/amdsw/omniperf/2.0.0-TP1/bin/utils/utils.py", line 46, in wrap_function
    result = function(*args, **kwargs)
  File "/autofs/nccs-svm1_sw/crusher/amdsw/omniperf/2.0.0-TP1/bin/omniperf_soc/soc_gfx90a.py", line 87, in profiling_setup
    self.perfmon_filter(self.get_args().roof_only)
  File "/autofs/nccs-svm1_sw/crusher/amdsw/omniperf/2.0.0-TP1/bin/utils/utils.py", line 46, in wrap_function
    result = function(*args, **kwargs)
  File "/autofs/nccs-svm1_sw/crusher/amdsw/omniperf/2.0.0-TP1/bin/omniperf_soc/soc_base.py", line 208, in perfmon_filter
    shutil.rmtree(self.__workload_dir)
  File "/opt/cray/pe/python/3.9.13.1/lib/python3.9/shutil.py", line 746, in rmtree
    onerror(os.path.islink, path, sys.exc_info())
  File "/opt/cray/pe/python/3.9.13.1/lib/python3.9/shutil.py", line 744, in rmtree
    raise OSError("Cannot call rmtree on a symbolic link")
```

**Comment by skyreflectedinmirrors** [comment]
To reproduce do e.g.,

```
omniperf profile -n vcopy -- ./vcopy <whatever>
cd workloads/
mkdir -p vcopy_copy
cd vcopy_copy
ln -s ../vcopy/MI* .
cd ..
cd ..
omniperf profile -n vcopy_copy -- ./vcopy
```

**Comment by skyreflectedinmirrors** [comment]
Just tested the following locally:

```
        # Initialize directories
        if not os.path.isdir(self.__workload_dir):
            os.makedirs(self.__workload_dir)
        elif not os.path.islink(self.__workload_dir):
            shutil.rmtree(self.__workload_dir)
        else:
            os.unlink(self.__workload_dir)
```

Seems to work

**Comment by coleramos425** [comment]
Closed in #297
